src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -4672,8 +4672,12 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
 	alias->colnames = NIL;
 
 	// get table alias
+	const CMDName *md_alias = (NULL != table_descr->MdAlias())
+								  ? table_descr->MdAlias()
+								  : table_descr->MdName();
+
 	alias->aliasname = CTranslatorUtils::CreateMultiByteCharStringFromWCString(
-		table_descr->MdName()->GetMDName()->GetBuffer());
+		md_alias->GetMDName()->GetBuffer());
```

Comment:
How about such optimization?
```c
if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
{
rte->alias = alias;
alias->aliasname = CTranslatorUtils::CreateMultiByteCharStringFromWCString(
		table_descr->MdAlias()->GetMDName()->GetBuffer());
}
else
{
alias->aliasname = CTranslatorUtils::CreateMultiByteCharStringFromWCString(
		table_descr->MdName()->GetMDName()->GetBuffer());
}
```

Reply:
Interesting... Is it better?

Reply:
Setting it dependently now

Reply:
I think, my solution is more simple and less invasive. Another solution:
```c
const CMDName *md_alias = table_descr->MdName();
if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
{
md_alias = table_descr->MdAlias();
rte->alias = alias;
}
alias->aliasname = CTranslatorUtils::CreateMultiByteCharStringFromWCString(
		md_alias->GetMDName()->GetBuffer());
```

Reply:
okay, removed ternary operator